bukkit/rpk-permissions-bukkit/src/main/kotlin/com/rpkit/permissions/bukkit/command/group/PermissionGroupListCommand.kt

```diff
@@ -0,0 +1,72 @@
+package com.rpkit.permissions.bukkit.command.group
+
+import com.rpkit.core.service.Services
+import com.rpkit.permissions.bukkit.RPKPermissionsBukkit
+import com.rpkit.permissions.bukkit.group.RPKGroupService
+import com.rpkit.players.bukkit.profile.RPKProfile
+import com.rpkit.players.bukkit.profile.minecraft.RPKMinecraftProfileService
+import org.bukkit.Bukkit
+import org.bukkit.command.Command
+import org.bukkit.command.CommandExecutor
+import org.bukkit.command.CommandSender
+import org.bukkit.entity.Player
+
+class PermissionGroupListCommand(private val plugin: RPKPermissionsBukkit) : CommandExecutor {
+
+    override fun onCommand(sender: CommandSender, command: Command, label: String, args: Array<String>): Boolean {
+        if (sender !is Player) {
+            sender.sendMessage(plugin.messages["not-from-console"])
+            return true
+        }
+        if (!sender.hasPermission("rpkit.permissions.command.character.group.list")) {
+            sender.sendMessage(plugin.messages["no-permission-character-group-list"])
+            return true
+        }
+
+        var playerUUID = sender.uniqueId;
+
+        if (args.isNotEmpty()) {
+            val player = Bukkit.getPlayer(args[0]);
+            if (player != null) {
+                playerUUID = Bukkit.getPlayer(args[0])!!.uniqueId;
```

Comment:
You don't need to retrieve the player again if you've already done so. Smart casting means you have a non-null ref to the player here.
```suggestion
                playerUUID = player.uniqueId;
```